src/storage/filestorage.hpp

```diff
@@ -122,6 +123,13 @@ class FileStorage : public aos::NonCopyable {
 
         aos::Buffer(record.mChecksum, aos::cSHA256Size) = checksum.mValue;
 
+        if (deletedRecordOffset > 0) {
```

Comment:
Will not work if the first record is deleted. Probably, `deletedRecordOffset` should be initialized with -1 if possible.